app/models/role.rb

```diff
@@ -145,6 +144,11 @@ def disable_filters_overriding
     self.filters.where(:override => true).map { |filter| filter.disable_overriding! }
   end
 
+  def clone
+    self.deep_clone(:except  => [:name, :builtin],
+                    :include => [:locations, :organizations, { :filters => :permissions }])
```

Comment:
would you mind updating https://github.com/theforeman/foreman/blob/develop/app/controllers/roles_controller.rb#L92 to use this `#clone` as well?